tentacle/src/transports/tls.rs

```diff
@@ -81,7 +81,9 @@ async fn connect(
         Some(socket_address) => {
             let stream = tcp_dial(socket_address, config.tls_bind, timeout).await?;
 
-            if let Some(domain_name) = parse_tls_domain_name(&addr) {
+            if let Some(domain_name) =
+                parse_tls_domain_name(&original.clone().unwrap_or_else(|| addr.clone()))
```

Comment:
parse_tls_domain_name(original.as_ref.unwrap_or(&addr))

Reply:
unwrap_or_else better than unwrap_or because of clippy

Reply:
My purpose is to delete clone